### ISSUE TITLE
A better way for dynamic mdns require for discovery

### ISSUE THREAD

**Issue by tkurki** [SignalK/signalk-js-client#23]
Discovery needs mdns, but the way it is handled as an optional dependency causes problems with webpack: https://github.com/SignalK/instrumentpanel/issues/58

**Comment by neftaly** [comment]
It might be worth tracking Mozilla @flyweb as an in-browser alternative to node mdns, though it'd require a (non-breaking) change to the mdns server. It'd be ideal to have a module that switched between mdns and a flyweb implementation, though that depends on how the spec pans out.

**Comment by timmathews** [comment]
Dependency injection seems like a worthwhile approach here. It's how I've implemented the API in #13.

Essentially, if you want to use mDNS, do something like this:

```js
import mdns from 'mdns';
import Client from '@signalk/signalk-client';

const skClient = new Client(null, mdns);
```

Passing `null` as the first parameter to the client constructor indicates that we want to get _all_ the configuration options from mDNS. It's also possible to override mDNS by passing a configuration object as the first param to the constructor.

**Comment by tkurki** [comment]
Agree!

**Comment by fabdrol** [comment]
This has been resolved with #41
